imjoy/helper.py

```diff
@@ -37,147 +40,245 @@ def killProcess(logger, pid):
     psutil = get_psutil()
     if psutil is None:
         return
+    logger.info("Killing plugin process (pid=%s)", pid)
     try:
         cp = psutil.Process(pid)
         for proc in cp.children(recursive=True):
             try:
                 if proc.is_running():
                     proc.kill()
             except psutil.NoSuchProcess:
-                logger.info("subprocess %s has already been killed", pid)
+                logger.info("Subprocess %s has already been killed", pid)
             except Exception as exc:  # pylint: disable=broad-except
                 logger.error(
-                    "WARNING: failed to kill a subprocess (PID={}). Error: {}".format(
-                        pid, str(exc)
-                    )
+                    "Failed to kill a subprocess (pid=%s). Error: %s", pid, exc
                 )
         cp.kill()
-        logger.info("plugin process %s was killed.", pid)
+        logger.info("Plugin process %s was killed.", pid)
     except psutil.NoSuchProcess:
-        logger.info("process %s has already been killed", pid)
+        logger.info("Process %s has already been killed", pid)
     except Exception as exc:  # pylint: disable=broad-except
         logger.error(
-            "WARNING: failed to kill a process (PID={}), "
-            "you may want to kill it manually. Error: {}".format(pid, str(exc))
+            "Failed to kill a process (pid=%s), "
+            "you may want to kill it manually. Error: %s",
+            pid,
+            exc,
         )
 
 
-def parseRequirements(requirements, default_command=None, conda=False):
-    """Parse requirements."""
-    if default_command is None:
-        default_command = ""
-    requirements_cmd = ""
-    if type(requirements) is list:
-        requirements = [str(r) for r in requirements]
-
-        for r in requirements:
-            if ":" in r:
-                rs = r.split(":")
-                tp, libs = rs[0], ":".join(rs[1:])
-                tp, libs = tp.strip(), libs.strip()
-                libs = [l.strip() for l in libs.split(" ") if l.strip() != ""]
-                if tp == "conda" and len(libs) > 0 and conda:
-                    requirements_cmd += " && conda install -y " + " ".join(libs)
-                elif tp == "pip" and len(libs) > 0:
-                    requirements_cmd += " && pip install " + " ".join(libs)
-                elif tp == "repo" and len(libs) > 0:
-                    pass
-                elif tp == "cmd" and len(libs) > 0:
-                    requirements_cmd += " && " + " ".join(libs)
-                elif "+" in tp or "http" in tp:
-                    requirements_cmd += " && pip install " + r
-                else:
-                    raise Exception("Unsupported requirement type: " + tp)
+def parse_reqs(reqs, conda=False):
```

Comment:
didn't we changed this to `parse_requirements` ? or this one is not rebased?

Reply:
Nah, it wasn't rebased. Now I just did that.